File: cpp/Pc.cpp
#include <bits/stdc++.h>
#include "Disk.cpp"
#include "Processor.cpp"
#include "Ram.cpp"

using namespace std;

class Pc
{
private:
    Disk disk;
    Processor processor;
    Ram ram;
    int totalPrice;

public:
    Pc()
    {
        this->disk = Disk();
        this->processor = Processor();
        this->ram = Ram();
        this->totalPrice = 0;
    }

    Pc(Disk disk, Processor processor, Ram ram)
    {
        this->disk = disk;
        this->processor = processor;
        this->ram = ram;
        this->totalPrice = disk.getPrice() + processor.getPrice() + ram.getPrice();
    }

    void setDisk(Disk disk)
    {
        this->disk = disk;
    }

    void setProcessor(Processor processor)
    {
        this->processor = processor;
    }

    void setRam(Ram ram)
    {
        this->ram = ram;
    }

    void setTotalPrice(int totalPrice)
    {
        this->totalPrice = totalPrice;
    }

    Disk getDisk()
    {
        return this->disk;
    }

    Processor getProcessor()
    {
        return this->processor;
    }

    Ram getRam()
    {
        return this->ram;
    }

    int getTotalPrice()
    {
        return this->totalPrice;
    }

    ~Pc(){};
};

File: java/Pc.java
package latihan3.java;

public class Pc {

    private Disk disk;
    private Processor processor;
    private Ram ram;
    private int totalPrice;

    public Pc() {
    }

    public Pc(Disk disk, Processor processor, Ram ram) {
        this.disk = disk;
        this.processor = processor;
        this.ram = ram;
        this.totalPrice = this.ram.getPrice() + this.processor.getPrice() + this.disk.getPrice();
    }

    public int getTotalPrice() {
        return totalPrice;
    }

    public void setTotalPrice(int totalPrice) {
        this.totalPrice = totalPrice;
    }

    public Disk getDisk() {
        return disk;
    }

    public void setDisk(Disk disk) {
        this.disk = disk;
    }

    public Processor getProcessor() {
        return processor;
    }

    public void setProcessor(Processor processor) {
        this.processor = processor;
    }

    public Ram getRam() {
        return ram;
    }

    public void setRam(Ram ram) {
        this.ram = ram;
    }

}
